core/libraries/form_sections/strategies/display/EE_License_Key_Display_Strategy.php

```diff
@@ -0,0 +1,33 @@
+<?php
+/* phpcs:disable PSR1.Classes.ClassDeclaration.MissingNamespace */
+/* phpcs:disable Squiz.Classes.ValidClassName.NotCamelCaps */
+/**
+ * EE_License_Key_Display_Strategy
+ * An input that receives a license key (and has a validation indicator)
+ *
+ * @package EventEspresso
+ * @author  Darren Ethier
+ * @since   $VID:$
+ */
+class EE_License_Key_Display_Strategy extends EE_Text_Input_Display_Strategy
+{
+    /**
+     * Whether or not the key has been verified (informs the style of the indicator)
+     * @var bool
+     */
+    private $verified;
+
+    public function __construct($verified = false)
+    {
+        $this->verified = $verified;
+        parent::__construct('text');
+    }
+
+    public function display()
+    {
+        $input = parent::display();
+        $verified_class = $this->verified ? 'ee-icon-color-ee-green' : 'ee-icon-color-ee-red';
+        $input .= '<span class="dashicons dashicons-admin-network ' . $verified_class . ' ee-icon-size-20"></span>';
+        return $input;
+    }
+}
```

Comment:
I don't think this extra class is necessary simply because it has no potential reusability beyond this one use case

I think a better more reusable solution would be to add a filter in one of the following places (all of which would allow you to do what you need):

- `EE_Admin_Two_Column_Layout::layout_input()` - not so great because only applies to this form layout
- `EE_Text_Input_Display_Strategy::display()` - not so great because only applies to this input layout
- `EE_Form_Input_Base::get_html_for_input()` - great because it would apply to all inputs and layouts

and then simply add your dashicon before the input html.

Another possibly more robust solution, but far more complex, would be to add dashicons to the form system as part of the `EE_Form_Input_Base` class and then pass your configuration details along when constructing the form. This of course would be best added via a separate issue.

For now I would recommend adding a filter to `EE_Form_Input_Base::get_html_for_input()`

Reply:
Your feedback prompted memory of something we already added to the form system that I had forgot about.  I can use the `VsprintfFilter` object for doing what's needed.